File: backend/src/main/java/vjda/repository/entity/ToDo.java
package vjda.repository.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Data;

/** A JPA entity corresponding to todo table */
@Entity
@Table(name="todo")
@Data
public class ToDo {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;
	private String title;
	private Category category;
	private String content;

	public enum Category { one, two, three }
}

File: backend/src/main/kotlin/backend/repository/entity/ToDo.kt
package backend.repository.entity

import javax.persistence.*

/** A JPA entity corresponding to todo table  */
// @Entity: Indicate that the class is a JPA entity.
@Entity
// @Table: Specifies the table name if it is not same as the class name.
@Table(name = "todo")
data class ToDo (
    // @Id: Indicates that the field uniquely identifies the entity.
    @Id
    // @GeneratedValue(strategy=GenerationType.IDENTITY):
    //   Generates the primary key by using the identity columns of RDBMS.
    //   (ex. MySQL: auto_increment)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var title: String? = null,
    var category: Category? = null,
    var content: String? = null
) {
    enum class Category {
        one, two, three
    }
}